src/lib389/lib389/perftools.py

```diff
@@ -0,0 +1,571 @@
+                "start_time" : first_time,
+                "stop_time" : last_time,
+                "nb_measures" : nbMes,
+                "rawresults" : rawres,
+                **self.getEnvInfo() }
+        return self.finalizeResult(res)
+
+    def mesure_export_import(self, nbMes=10):
+        self._instance.stop()
+        self._ldif = self.getFilePath("db.ldif");
+        self._nbEntries = None
+        res = [ self._do_measure("export", self.offline_export, nbMes), self._do_measure("import", self.offline_import, nbMes) ]
+        self._instance.start()
+        return res;
+
+    class Tester:
+        # Basic tester (used to define ldclt tests)
+        def __init__(self, name, description, method_name):
+            self._base_name = name
+            self._base_description = description
+            self._method_name = method_name
+
+        def name(self):
+            return self._base_name
+
+        def argsused(self):
+            return [ "nb_threads", "name" ]
+        
+        def description(self):
+            return self._base_description
+
+        def run(self, perftools, args):
+            args['name'] = self._base_name
+            res = getattr(perftools, self._method_name)(self._base_name, nb_threads=args['nb_threads']);
+            print (res['pretty'])
+
+        @staticmethod
+        def initTester(args):
+            os.environ["NSSLAPD_DB_LIB"] = args['db_lib']
+            perftools = PerformanceTools( args )
+            perftools.initInstance()
+            return perftools;
+
+    class TesterImportExport(Tester):
+        # A special tester for export/import
+        def __init__(self):
+            super().__init__("export/import", 
+                "Measure export rate in entries per seconds then measure import rate.",
+                 None)
+
+        def argsused(self):
+            return []
+        
+        def run(self, perftools, args=None):
+            res = perftools.mesure_export_import()
+            for r in res:
+                print (r['pretty'])
+
+    class TesterAll(Tester):
+        # One to run them all
+        def __init__(self, testers):
+            super().__init__("all", 
+                "Run all tests for 100,1K,10K,100K,1M users with both db lib with 1, 4 and 8 threads.",
+                 None)
+            self._testers = testers
+
+        def argsused(self):
+            return []
+        
+        def run(self, perftools, args=None):
+            for users in ( 100, 1000, 10000, 100000, 1000000 ):
+                for db in ( 'bdb', 'mdb' ):
+                    perftools = PerformanceTools.Tester.initTester({'nbUsers': users, 'db_lib': db})
+                    for t in self._testers:
+                        if 'nb_threads' in t.argsused():
+                            for nbthreads in ( 1, 4, 8 ):
+                                t.run(perftools, { "nb_threads" : nbthreads })
+                        else:
+                            t.run(perftools)
```

Comment:
Implementing the "dsrate run all" feature as a Tester make things a bit cumbersome (run  parameters are ignored and instance is generated twice for first test)  should may have implement a runall sub command backed on a runAllTests static method